src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java

```diff
@@ -53,4 +81,105 @@ public NewConnectionNavigationPage(String title,
 
       return pages;
    }
+
+   private static Widget createWidget(ArrayList<WizardPage<NewConnectionContext, ConnectionOptions>> pages)
+   {
+      return new Selector(pages);
+   }
+
+   private static class Selector
+         extends Composite
+         implements HasWizardPageSelectionHandler<NewConnectionContext, ConnectionOptions>
+   {
+      public Selector(final ArrayList<WizardPage<NewConnectionContext, ConnectionOptions>> pages)
+      {
+         WizardResources.Styles styles = WizardResources.INSTANCE.styles();
+         
+         ScrollPanel scrollPanel = new ScrollPanel();
+         scrollPanel.setSize("100%", "100%");
+         scrollPanel.addStyleName(styles.wizardPageSelector());
+
+         VerticalPanel verticalPanel = new VerticalPanel();
+         verticalPanel.setSize("100%", "100%");
+
+         for (int i = 0, n = pages.size(); i < n; i++)
+         {
+            final WizardPage<NewConnectionContext, ConnectionOptions> page = pages.get(i);
+            SelectorItem item = new SelectorItem(page, new ClickHandler()
+            {
+               @Override
+               public void onClick(ClickEvent event)
+               {
+                  onSelected_.execute(page);
```

Comment:
Not really, `setSelectionHandler` is called during `addAndInitializePage` while constructing the page before any click event would reach this. Also, the eisting code does not check for this, something bad might be going on is this is actually NULL.